crates/miden-objects/src/note/assets.rs

```diff
@@ -107,6 +108,43 @@ impl NoteAssets {
         padded_assets
     }
 
+    /// Returns all Fungible Assets as an Iterator.
+    pub fn iter_fungible(&self) -> impl Iterator<Item = FungibleAsset> {
+        self.assets.iter().filter_map(|asset| {
+            if asset.is_fungible() {
+                Some(asset.unwrap_fungible())
+            } else {
+                None
+            }
+        })
```

Comment:
```suggestion
         self.assets.iter().filter_map(|asset| match asset {
                Asset::Fungible(fungible_asset) => Some(*fungible_asset),
                Asset::NonFungible(_) => None,
            })
        })
```

We can avoid two method calls by matching. Should be more efficient. We can rewrite `iter_non_fungible` in the same way.